src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedTryResourceShouldBeUnnamedCheck.java

```diff
@@ -0,0 +1,260 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code and other text files for adherence to a set of rules.
+// Copyright (C) 2001-2025 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.coding;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.List;
+import java.util.Optional;
+
+import com.puppycrawl.tools.checkstyle.FileStatefulCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * <div>
+ * Ensures that try resources that are not used are declared as an unnamed variable.
+ * </div>
+ *
+ * <p>
+ * Rationale:
+ * </p>
+ * <ul>
+ *     <li>
+ *         Improves code readability by clearly indicating which resources are unused.
+ *     </li>
+ *     <li>
+ *         Follows Java conventions for denoting unused parameters with an underscore ({@code _}).
+ *     </li>
+ * </ul>
+ *
+ * <p>
+ * See the <a href="https://docs.oracle.com/en/java/javase/21/docs/specs/unnamed-jls.html">
+ * Java Language Specification</a> for more information about unnamed variables.
+ * </p>
+ *
+ * <p>
+ * <b>Attention</b>: Unnamed variables are available as a preview feature in Java 21,
+ * and became an official part of the language in Java 22.
+ * This check should be activated only on source code which meets those requirements.
+ * </p>
+ *
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ *
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code unused.try.resource}
+ * </li>
+ * </ul>
+ *
+ * @since 12.2.0
+ */
+@FileStatefulCheck
```

Comment:
If possible, it will be good to make it state less.
It will help a lot during migration to multi threading or checkstyle